appserver/persistence/cmp/support-sqlstore/src/main/java/com/sun/jdo/spi/persistence/support/sqlstore/connection/ConnectionImpl.java

```diff
@@ -871,13 +873,14 @@ public synchronized String toString() {
         return buffer;
     }
 
-    protected void finalize() {
-        try {
-            this.connection.close();
-            logger.finest("sqlstore.connectionimpl.finalize"); // NOI18N
-        } catch (SQLException se) {
-            ;
-        }
+    public final void registerCloseEvent() {
+        Cleaner.create().register(this, () -> {
```

Comment:
Each call to Cleaner.create() creates one new daemon thread internally. Could this create too many threads? The example from the Javadoc uses one single static Cleaner instance to register objects with and uses the returned Cleanable instance returned from the register call to clean.